File: server/models/MessageModel.js
const { v4: uuidv4 } = require('uuid');

class MessageModel {
    constructor(roomId, senderId, senderName, content) {
        this.id = uuidv4();
        this.roomId = roomId;
        this.senderId = senderId;
        this.senderName = senderName;
        this.content = content;
        this.timestamp = new Date();
    }
}

module.exports = MessageModel;

File: lib/data/models/message_model.dart
// 消息类型枚举
enum MessageType { text, system, image }

class MessageModel {
  final String id;
  final String roomId;
  final String senderId;
  final String senderName;
  final String content;
  final DateTime timestamp;
  final MessageType type;

  MessageModel({
    required this.id,
    required this.roomId,
    required this.senderId,
    required this.senderName,
    required this.content,
    required this.timestamp,
    this.type = MessageType.text,
  });
}
